### ISSUE TITLE
Bidirectional DSHOT

### ISSUE THREAD

**Issue by bkleiner** [FETtec/ESC-Firmware#3]
Hi,
Is there a chance that bidirectional dshot will be supported eventually?
Really like the hardware so far :+1:

**Comment by IB1387** [comment]
Bump. The only thing missing from this is support for RPM filtering. :-)

**Comment by speurhond** [comment]
RPM filtering would be great

**Comment by wuzle** [comment]
I agree. It would not have to be bidirectional DShot though, it could also be implemented into Betaflight. While I don't have any clue on how much work it actually is to make Betaflight OneWire compatible, in my opinion the update speed should be enough to provide a 8kHz PID-Loop with sufficient RPM-data.

In general I have not flown an ESC that so finely controls motors like the Fettec 35A. It really makes a big difference on tight lines and difficult tracks. This weekend I had the chance to directly compare two identical setups (one Singularitum with a Fettec35A V1.0 and another with that 20x20 Hobbywing BLHeli32) on our training track and the difference is quite stunning. The Fettec runs way smoother all around and provides way more control especially on the lower side of the throttle curve and upps the efficiency quite a bit. The only thing missing is the RPM-Filter. With that enabled you'd have no vibration leaking into the gyro data whatsoever with minimal filter latencies and I can't imagine that there is anything comparable on the market at the moment.
Right now, the Fettec ESC is slightly above my BLHElis32 in noise profiles, RPM-Filters would -I guess- catapult it way ahead.

**Comment by stackola** [comment]
Bump

Can we get any update from the author on this?

**Comment by lichtl** [comment]
There is no plan to implement bidirectional DSHOT. The best is if Betaflight would directly support the Onewire protocol (which then would allow to get all telemetry data in real time (voltage/current/rpm).

**Comment by stackola** [comment]
Are you guys working on getting Onewire into BTFL, or do you just hope someone from BTFL will do it them self?

**Comment by wuzle** [comment]
I guess Onewire integration into Betaflight would be even better. In the end for the consumer it doesn't really matter where the juice comes from :)

There is also additional functionalities like Battery compensation features that we could enable when there's fast enough telemetry data (normal telemetry is not fast enough)

**Comment by bkleiner** [comment]
> There is no plan to implement bidirectional DSHOT. The best is if Betaflight would directly support the Onewire protocol (which then would allow to get all telemetry data in real time (voltage/current/rpm).

Is there a specification document for OneWire (i guess its "just" serial?) to get the ball rolling on this.

**Comment by rauar** [comment]
There's a (probably very old) document on rcgroups - unclear how much it fits the current onewire implementation on fettec escs: https://www.rcgroups.com/forums/showatt.php?attachmentid=8524039&d=1450424877

It mentions a PWM-based trigger to receive telemetry...

@lichtl: Would be great if the fettec devs could share some specs so that there's a chance to get it implemented in Betaflight. Currently stuck without RPM filter using a Betaflight FC and a Fettec 4in1 😖

**Comment by lichtl** [comment]
Hi @rauar 

please find attached 2 implementations (one for Arduino and the one used in KISS FC firmware). They do contain some documentation as well. Hope this helps for the start. 

[FETtecOneWire.zip](https://github.com/FETtec/ESC-Firmware/files/5414718/FETtecOneWire.zip)
[FEttecESC_arduino.zip](https://github.com/FETtec/ESC-Firmware/files/5414719/FEttecESC_arduino.zip)

**Comment by stackola** [comment]
@lichtl Are you guys actively working on getting this into BTFL, or are you just hoping and waiting for some community member to get around to that?

**Comment by rauar** [comment]
@lichtl thanks a lot for the files. will check them out as soon as i find some spare time.

@stackola check out this betaflight issue - i tried to reactive the discussion on a former request and the betaflight and blheli devs plus Felix from Fettec were commenting: https://github.com/betaflight/betaflight/pull/7264#issuecomment-713170259

Unfortunately I seems like there are 2 ways how to implement this and there's no agreement yet on one of those options. One option is to implement it in DSHOT on the signal line (like BLHeli) and Mike prefers this (avoid multiple implementations of the same "thing") and Felix seems to promote Oneshot (using a dedicated telemetry wire).

Not sure if they will agree on one of those options (I doubt it atm) - but I see the points of both sides.

I think the more people ask for support for RPM filtering when using Kiss/Fettec ESCs the higher the chances they come to an agreement.

**Comment by Linjieqiang** [comment]
@rauar Any news?

**Comment by Linjieqiang** [comment]
@ronlix Any docs about the Fettec Onewire?

**Comment by stackola** [comment]
It's been almost a year since you have claimed to be going into testing. @FETtec 

Can your customers ever expect to get working RPM filters on betaflight or not?

**Comment by krunked-fpv** [comment]
Bruh. Lol. Give it up.

**Comment by stackola** [comment]
> Bruh. Lol. Give it up.

No!

Bump.
Give me bidrectional dshot!!!

**Comment by stackola** [comment]
Any day now!

**Comment by krunked-fpv** [comment]
Give it up. Lol

**Comment by VolkerGoeschl** [comment]
Never gonna give you up, never gonna let you down, never gonna run around to D-shot!

**Comment by stackola** [comment]
> Never gonna give you up, never gonna let you down, never gonna run around to D-shot!

Found a fix: Sold my Fettec stuff and won't be buying any more.

**Comment by homepods** [comment]
with or without bidirectional dshot - does it matter? perhaps no. if you test side by side BF with rpm filtering/bidi shot and FT would you see a big difference or any difference at all?

**Comment by stackola** [comment]
@homepods It matters a lot. AFAIK, FT also uses some bidirectional protocol but it's homebrew and not compatible with BF.

**Comment by homepods** [comment]
> @homepods It matters a lot. AFAIK, FT also uses some bidirectional protocol but it's homebrew and not compatible with BF.

As I understand FT reads through the TLM the motor rpm and does the filtering accordingly. 

- BLheli 32 128K esc -> 24khz to ByRPM
- Dshot 1200 or max supported
- Fettech Alpha with G4 1.7 N

would you keep a fix pwm frequency or variable?

**Comment by d8vit** [comment]
It's been 5 years and still the developers don't give a damn about us? This is essentially one software change! Because of your stupid war and lack of compatibility you just lost a customer!
